Tests/UnitTests/Core/TrackFitting/Gx2fTests.cpp

```diff
@@ -0,0 +1,352 @@
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/EventData/VectorMultiTrajectory.hpp"
+#include "Acts/EventData/VectorTrackContainer.hpp"
+#include "Acts/Geometry/CuboidVolumeBuilder.hpp"
+#include "Acts/Geometry/Layer.hpp"
+#include "Acts/Geometry/TrackingGeometry.hpp"
+#include "Acts/Geometry/TrackingGeometryBuilder.hpp"
+#include "Acts/MagneticField/ConstantBField.hpp"
+#include "Acts/Material/HomogeneousSurfaceMaterial.hpp"
+#include "Acts/Material/HomogeneousVolumeMaterial.hpp"
+#include "Acts/Material/MaterialSlab.hpp"
+#include "Acts/Propagator/EigenStepper.hpp"
+#include "Acts/Propagator/Navigator.hpp"
+#include "Acts/Propagator/Propagator.hpp"
+#include "Acts/Propagator/StraightLineStepper.hpp"
+#include "Acts/Surfaces/RectangleBounds.hpp"
+#include "Acts/Tests/CommonHelpers/DetectorElementStub.hpp"
+#include "Acts/Tests/CommonHelpers/MeasurementsCreator.hpp"
+#include "Acts/Tests/CommonHelpers/PredefinedMaterials.hpp"
+#include "Acts/TrackFitting/GainMatrixUpdater.hpp"
+#include "Acts/TrackFitting/GlobalChiSquareFitter.hpp"
+
+#include <vector>
+
+#include "FitterTestsCommon.hpp"
+
+using namespace Acts::UnitLiterals;
+
+namespace Acts {
+namespace Test {
+
+//// Construct initial track parameters.
+Acts::CurvilinearTrackParameters makeParameters(
+    const ActsScalar x = 0.0_m, const ActsScalar y = 0.0_m,
+    const ActsScalar z = 0.0_m, const ActsScalar w = 42_ns,
+    const ActsScalar phi = 0_degree, const ActsScalar theta = 90_degree,
+    const ActsScalar p = 1_GeV, const ActsScalar q = 1_e) {
+  // create covariance matrix from reasonable standard deviations
+  Acts::BoundVector stddev;
+  stddev[Acts::eBoundLoc0] = 100_um;
+  stddev[Acts::eBoundLoc1] = 100_um;
+  stddev[Acts::eBoundTime] = 25_ns;
+  stddev[Acts::eBoundPhi] = 2_degree;
+  stddev[Acts::eBoundTheta] = 2_degree;
+  stddev[Acts::eBoundQOverP] = 1 / 100_GeV;
+  Acts::BoundSymMatrix cov = stddev.cwiseProduct(stddev).asDiagonal();
+  // define a track in the transverse plane along x
+  Acts::Vector4 mPos4(x, y, z, w);
+  return Acts::CurvilinearTrackParameters(mPos4, phi, theta, p, q, cov);
+}
+
+// Construct a straight-line propagator.
+auto makeStraightPropagator(std::shared_ptr<const Acts::TrackingGeometry> geo) {
+  Acts::Navigator::Config cfg{std::move(geo)};
+  cfg.resolvePassive = false;
+  cfg.resolveMaterial = true;
+  cfg.resolveSensitive = true;
+  Acts::Navigator navigator(cfg);
+  Acts::StraightLineStepper stepper;
+  return Acts::Propagator<Acts::StraightLineStepper, Acts::Navigator>(
+      stepper, std::move(navigator));
+}
+
+static std::vector<Acts::SourceLink> prepareSourceLinks(
+    const std::vector<TestSourceLink>& sourceLinks) {
+  std::vector<Acts::SourceLink> result;
+  std::transform(sourceLinks.begin(), sourceLinks.end(),
+                 std::back_inserter(result),
+                 [](const auto& sl) { return Acts::SourceLink{sl}; });
+  return result;
+}
+
+std::shared_ptr<const TrackingGeometry> makeToyDetector(
```

Comment:
You could consider also using this functionality, it seems to be quite similar: [Tests/CommonHelpers/Acts/Tests/CommonHelpers/CubicTrackingGeometry.hpp](https://github.com/acts-project/acts/blob/main/Tests/CommonHelpers/Acts/Tests/CommonHelpers/CubicTrackingGeometry.hpp)

But you can also stick to this for now I think.